cpp/src/arrow/python/numpy_to_arrow.cc

```diff
@@ -743,7 +743,9 @@ Status NumPyConverter::ConvertDecimals() {
 
   if (type_ == NULLPTR) {
     for (PyObject* object : objects) {
-      RETURN_NOT_OK(max_decimal_metadata.Update(object));
+      if (!internal::PandasObjectIsNull(object)) {
```

Comment:
Do we care about accepting other NULL-like objects such as `float('nan')`? Otherwise `object != Py_None` is a much faster check.

Reply:
I'm not sure, is it possible to get NaNs from operations on Decimals?  Or is that something the user might mix in somehow?

Reply:
Python decimal objects can be nan, unfortunately:

```python
>>> import decimal
>>> decimal.Decimal('nan')
Decimal('NaN')
```

Reply:
Seems like it could be NaN also:
```
In [5]: s1 = pd.Series([Decimal('1.0'), Decimal('2.0')])

In [6]: s2 = pd.Series([Decimal('2.0'), None])

In [7]: s1 / s2
Out[7]: 
0    0.5
1    NaN
dtype: object